tests/containers/podman_pods.pm

```diff
@@ -83,12 +84,31 @@ sub run {
             install_k3s();
             record_info('Test', 'kube apply');
             assert_script_run('podman kube apply --kubeconfig ~/.kube/config -f pod.yaml');
-            assert_script_run('kubectl wait --for=condition=Ready pod/testing-pod');
+            assert_script_run('kubectl wait --for=condition=Ready pod/testing-pod', timeout => 240);
             validate_script_output('kubectl exec testing-pod -- cat /etc/os-release', sub { m/SUSE Linux Enterprise Server/ });
         }
     }
+}
+
+sub cleanup {
+    my ($self) = @_;
+    $self->{podman}->cleanup_system_host();
+    uninstall_k3s();
+}
+
+sub post_run_hook {
+    my ($self) = @_;
+    $self->cleanup();
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+    $self->cleanup();
+}
 
-    $podman->cleanup_system_host();
+sub test_flags {
+    # Always rollback due to k3s
+    return {always_rollback => 1};
```

Comment:
Why? you are doing a cleanup anyway

Reply:
Pavel's idea was to ensure that the VM is in a well-defined state and we believe that installing `k3s` is rather intrusive. So, rolling back to a state where `k3s` is not installed might be desirable to prevent side-effects from that installation.

Reply:
Removed. Thanks for the hint Martin.